text/batch-l2.md

```diff
@@ -0,0 +1,1540 @@
+
+```ts
+import * as batch from 'aws-cdk-lib/aws-batch';
+
+new batch.MangedEC2ComputeEnvironment(this, 'myEc2ComputeEnv', {
+   spot: true,
+  spotBidPercentage: 20,
+});
+```
+
+You can also use Fargate spot instances, but you can’t specify `spotBidPercentage`:
+
+```ts
+new batch.FargateComputeEnvironment(this, 'myFargateComputeEnv', {
+   spot: true,
+});
+```
+
+For stateful or otherwise non-interruption-tolerant workflows, omit `spot` or set it to `false` to only provision on-demand instances. 
+
+#### Allocation Strategies
+
+Batch provides different Allocation Strategies to help it choose which instances to provision. 
+If you are using spot instances, you can choose `AllocationStrategy.SPOT_CAPACITY_OPTIMIZED`; 
+this will tell Batch to choose the instance types from the ones you’ve specified that have 
+the most spot capacity available to minimize the chance of interruption. 
+This means that to get the most benefit from your spot instances, 
+you should allow Batch to choose from as many different instance types as possible. 
+If you enable `spot` on your `ComputeEnvironment`, the allocation strategy will default to `SPOT_CAPACITY_OPTIMIZED`.
+This example configures your `ComputeEnvironment` to use several different types of instances:
+
+```ts
+const computeEnv = new batch.ManagedEC2ComputeEnvironment(this, 'myEc2ComputeEnv', {
+  spot: true,
+  instanceTypes: [batch.InstanceType.of(ec2.InstanceClass.M5AD, ec2.InstanceSize.LARGE)],
+});
+```
+
+Batch allows you to specify only the instance class and to let it choose the size, which you can do like this:
+
+```
+computeEnv.addInstanceClass(ec2.InstanceClass.M5AD);
+
+// Or, specify it on the constructor:
+const computeEnv = new batch.ManagedEC2ComputeEnvironment(this, 'myEc2ComputeEnv', {
+  spot: true,
+  instanceClasses: [ec2.InstanceClass.A1],
+});
+```
+
+If you specify no instance types, then it will default to `InstanceType.OPTIMAL`, which tells Batch to pick an instance from the C4, M4, and R4 instance families.
+You can specify both `InstanceType.OPTIMAL` alongside several different instance types in the same compute environment:
+
+```ts
+computeEnv.addInstanceType(batch.InstanceType.OPTIMAL);
+
+Note: this is equivalent to specifying
+
+computeEnv.addInstanceClass(ec2.InstanceClass.C4);
+computeEnv.addInstanceClass(ec2.InstanceClass.M4);
+computeEnv.addInstanceClass(ec2.InstanceClass.R4);
+```
+
+If your workflow does not tolerate interruptions and you want to minimize your costs, use `AllocationStrategy.BEST_FIT`.
```

Comment:
Introduce the tradeoff here as well. Right now this is phrased as if it doesn't have downsides.

```suggestion
If your workflow does not tolerate interruptions but it *does* tolerate delay and you want to minimize your costs, use `AllocationStrategy.BEST_FIT`.
```

Reply:
I wonder if all this prose about allocation strategies could be summarized more scannably in a table:

| Strategy | Optimized for | Downsides | Notes |
|-------|----------|----------|---------|
| BEST_FIT | Cost | May lower throughput | |
| BEST_FIT_PROGRESSIVE | Throughput | May increase cost | |
| SPOT_CAPACITY_OPTIMIZED | Least interruption | (None?) | Spot only |

etc.

Reply:
I've opted to add the table for conciseness, but I still think the explanations of what they actually mean are worth while.